gptel-openai.el

```diff
@@ -73,16 +74,24 @@
     (apply #'concat (nreverse content-strs))))
 
 (cl-defmethod gptel--parse-response ((_backend gptel-openai) response _info)
-  (map-nested-elt response '(:choices 0 :message :content)))
+  ;; If the reply specifies a function call, parse and return it instead of the message
+  (let* ((choices-path '(:choices 0 :message))
+         (tool-calls (map-nested-elt response (append choices-path '(:tool_calls))))
+         (content (map-nested-elt response (append choices-path '(:content)))))
+    (if tool-calls
+        (prin1-to-string tool-calls)
+      content)))
 
 (cl-defmethod gptel--request-data ((_backend gptel-openai) prompts)
   "JSON encode PROMPTS for sending to ChatGPT."
   (let ((prompts-plist
          `(:model ,gptel-model
            :messages [,@prompts]
            :stream ,(or (and gptel-stream gptel-use-curl
-                         (gptel-backend-stream gptel-backend))
-                     :json-false))))
+                             (gptel-backend-stream gptel-backend))
+                        :json-false))))
+    (when gptel-callable-functions
+      (plist-put prompts-plist :tools gptel-callable-functions))
```

Comment:
This is the main important change. This could be toggled on or off based on a setting in the transient. A certain function call can also be forced by using the `tool_choice` parameter